tools/crm_error.c

```diff
@@ -57,11 +56,99 @@ static GOptionEntry entries[] = {
     { NULL }
 };
 
+PCMK__OUTPUT_ARGS("result-code", "int", "char *", "char *");
+static int
+result_code_default(pcmk__output_t *out, va_list args)
+{
+    int code = va_arg(args, int);
+    char *name = va_arg(args, char *);
+    char *desc = va_arg(args, char *);
+
+    long long most_negative = 0;
+    int code_width = 0;
+
+    if (out->is_quiet(out)) {
+        /* If out->is_quiet(), don't print the code. Print name and/or desc in a
+         * compact format for text output, or print nothing at all for none-type
+         * output.
+         */
+        if (pcmk__str_eq(out->fmt_name, "text", pcmk__str_none)) {
```

Comment:
We only support `text`, `xml`, and `none`, so I think our options are:
* What I've got right now
* Have separate, probably mostly redundant formatters for `default` (`none`) and `text` to avoid this special handling
* Get rid of the `default` formatter altogether and convert this to a `text` formatter without the special check. I think you mentioned in a separate PR that, questions of style aside, if an output type doesn't have a formatter registered for a particular message then the message is just ignored.

Reply:
The last option sounds good